hecuba_py/src/hdict.py

```diff
@@ -268,26 +267,44 @@ def __init__(self, name=None, primary_keys=None, columns=None, tokens=None,
         else:
             self._primary_keys = primary_keys
             set_pks = []
+            normal_columns = []
             for column_name, column_type in columns:
                 if column_name.find("_set_") != -1:
                     set_pks.append((column_name.replace("_set_", ""), column_type))
+                else:
+                    normal_columns.append((column_name, column_type))
             if set_pks:
-                self._columns = [{"type": "set", "primary_keys": set_pks}]
+                self._columns = [{"type": "set", "columns": set_pks}]
             else:
                 self._columns = columns
             self._indexed_on = indexed_on
 
         self._has_embedded_set = False
-        for attr in self._columns:
-            if isinstance(attr, dict) and "type" in attr and attr["type"] == "set":
-                self._has_embedded_set = True
-                set_types = attr["primary_keys"]
-                self._build_column = []
-                for set_type in set_types:
-                    self._build_column.append(("_set_" + set_type[0], set_type[1]))
-
-        key_names = [pkname for (pkname, dt) in self._primary_keys]
-        column_names = [colname for (colname, dt) in self._columns]
+        self._build_column = []
+        columns = []
+        for col in self._columns:
+            if isinstance(col, dict) and "type" in col:
```

Comment:
What if is a dict but type is not in col?
Goes to the else and access a dictionary by ints, which will raise KeyError, and crashes I guess. Think about the conditions :)
@adrianespejo